### DISCUSSION TITLE
GH upload-artifact fails consistently when run in a container

### DISCUSSION THREAD

**Discussion by itsmemarioss** [community/community#67740] [Actions] [answered]
### Select Topic Area

Bug

### Body

The upload-artifact action started to hang up and fail apparently for no reason with the following error

```
Create Artifact Container - Attempt 1 of 5 failed with error: read ECONNRESET
Create Artifact Container - Attempt 2 of 5 failed with error: read ECONNRESET
```

I have noticed that it only happens if the step is run inside a container:

```
runs-on: []
container:
...
```

without the container, it will work. I also tried with different images, so at this point, I don't think the problem comes from my docker image.

The strange thing about this is that it was working a few days ago and just stopped.

It was also noticed that the action hangs up in this request: https://pipelinesghubeus14.actions.githubusercontent.com/<some hash>/_apis/pipelines/workflows/<id>/artifacts?api-version=6.0-preview

and the response was

```
{
  "$id": "1",
  "innerException": null,
  "message": "The user 'System:PublicAccess;aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' is not authorized to access this resource.",
  "typeName": "Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server",
  "typeKey": "UnauthorizedRequestException",
  "errorCode": 0,
  "eventId": 3000
}
```

The only workaround I have found was downgrading the upload-artifact to v1. What made me more confused about the whole thing.

Does anyone have any idea of what may be the problem?

**Comment by itsmemarioss** [comment, marked as answer]
In case someone faces the same issue or something similar, the problem came from a docker configuration. Docker’s MTU was updated on the runners. Not sure why the v1 works, probably it produces a different payload.

**Reply by serene-capybara** [reply]
Did you manage to find a solution involving the docker configuration (without downgrading the action version)?

**Reply by itsmemarioss** [reply]
The MTU configuration was modified by the DevOps team, but I lack specific information about the changes made. It seems that the containers may have been initialized with an incorrect value, potentially exceeding the supported value of the virtual machine. After the fix, the pipeline worked as expected with the new version.
